cmd/qat_plugin/README.md

```diff
@@ -1,39 +1,215 @@
+- [Intel&reg; Communications Chipset 8900 to 8920 Series][4]
+- [Intel&reg; Atom&trade; Processor C2000][5]
+
+> **FIXME** - we need to verify this list:
+> - do se support C2000
+> - do we need to add D15xx devices
+
+The QAT device plugin provides access to QAT hardware accelerated cryptographic and compression features.
+Demonstrations are provided utilising [DPDK](https://doc.dpdk.org/) and [OpenSSL](https://www.openssl.org/).
+
+[Kata Containers](https://katacontainers.io/) QAT integration is documented in the
+[Kata Containers documentation repository][6].
+
+## Modes and Configuration options
+
+The QAT plugin can take a number of command line arguments, summarised in the following table:
+
+| Argument | Meaning |
+|:-------- |:------- |
+| -debug | enable debug output
+| -dpdk-driver string | DPDK Device driver for configuring the QAT device (default "vfio-pci") |
+| -kernel-vf-drivers string | Comma separated VF Device Driver of the QuickAssist Devices in the system. Devices supported: DH895xCC,C62x,C3xxx and D15xx (default "dh895xccvf,c6xxvf,c3xxxvf,d15xxvf") |
+| -max-num-devices int | maximum number of QAT devices to be provided to the QuickAssist device plugin (default 32) |
+| -mode string | plugin mode which can be either dpdk (default) or kernel (default "dpdk") |
+
+The example [DaemonSet YAML](../../deployments/qat_plugin/qat_plugin.yaml) passes a number of these
+arguments, and takes its default values from the
+[QAT default ConfigMap](../../deployments/qat_plugin/qat_plugin_default_configmap.yaml). The following
+table summarises the defaults:
+
+| Argument | Variable | Default setting | Explanation |
+|:-------- |:-------- |:--------------- |:----------- |
+| -dpdk-driver | `$DPDK_DRIVER` | vfio-pci | A more robust and secure choice than the `igb_uio` alternative |
+| -kernel-vf-drivers | `$KERNEL_VF_DRIVERS` | dh895xccvf,c6xxvf,c3xxxvf,d15xxvf | Modify to suit your hardware setup |
+| -max-num-devices | `$MAX_NUM_DEVICES` | 32 | Modify to suit your hardware setup if necessary |
+| -debug | `$DEBUG` | false | Debug is disabled by default |
+
+For more details on the `-dpdk-driver` choice, see 
+[DPDK Linux Driver Guide](http://dpdk.org/doc/guides/linux_gsg/linux_drivers.html).
+
+> **FIXME** - I could not find any information around `kernel-vf-drivers` on the following site:
+> 
+> Use the `kernel-vf-drivers flag` to specify the vf Device Driver for the particular QAT device. For more information, refer to [Intel QAT Crypto Poll Mode Driver](https://dpdk.org/doc/guides/cryptodevs/qat.html).
+
+If the `-mode` parameter is set to `kerneldrv`, no other parameter documented above are valid,
+except `-debug` which is global for both modes.
+`kerneldrv` mode implements resource allocation based on system configured [logical instances][7].
+
+The `kerneldrv` mode does not guarantee full device isolation between containers
+and therefore it's not recommended. This mode will be deprecated and removed once `libqat`
```

Comment:
the default build also disables this mode.

Reply:
Ah, yes, I had some mixed understanding and wording. Have clarified both here and in the 'build' section.

Reply:
should we add a note here that the pre-built images do not have this option?

Reply:
done